dist/inverted-index.js

```diff
@@ -1,230 +0,0 @@
-      return bookKeys.length === 2 && bookKeys.includes('title') && book.title && typeof book.title === 'string' && bookKeys.includes('text') && book.text && typeof book.text === 'string' ? acc : (acc.push(bookIndex), acc);
-    }
-
-    /**
-     * normalize and tokenize book contents
-     * @param {Object} book -Book object
-     * @param {string} book.title -title of the book
-     * @param {string} book.text -text of the book
-     * @return {array.<string>} -an arry of words in book
-     */
-
-  }, {
-    key: 'analyse',
-    value: function analyse(book) {
-      // merge title and text of the object
-      var bookText = book.title + ' ' + book.text;
-      // change all non-word characters including '_' to ' '
-      var normalizedBookText = bookText.replace(/\W+|_+/g, ' ').trim().toLowerCase();
-      // tokenize; split normalized book text to word units
-      var bookTokens = normalizedBookText.split(/\s+/);
-      return bookTokens;
-    }
-
-    /**
-     * flattens array
-     * @param {array} nestedArray -A multidimensional array
-     * @returns {array} -A one dimensional array
-     */
-
-  }, {
-    key: 'flattenArray',
-    value: function flattenArray(nestedArray) {
-      var _this2 = this;
-
-      nestedArray.reduce(function (arr, item) {
-        return arr.concat(Array.isArray(item) ? _this2.flattenArray(item) : item);
-      }, []);
-    }
-
-    /**
-     * Search for occurence of two or more words in a book
-     * @param {Object} index - Index to be searched
-     * @param {string} chainedTerms - A space delimited string
-     *  of two or more words
-     * @returns {array} -An array of books where terms occur simultaneously
-     */
-
-  }, {
-    key: 'multiTermSearch',
-    value: function multiTermSearch(index, chainedTerms) {
-      // 'we are here' to ['we', 'are', 'here]
-      var termsArray = chainedTerms.split(' ');
-      // Accumulate indexes of all terms
-      // [[0, 1], [4, 5], [5, 1]] to [0, 1, 4, 5, 5, 1]
-      var indicesArray = termsArray.reduce(function (acc, term) {
-        return acc.push.apply(acc, _toConsumableArray(index[term]));
-      }, []);
-      var indexCount = {};
-      // count the number of occurence of each index in indicesArray
-      indicesArray.forEach(function (val) {
-        Object.prototype.hasOwnProperty.call(indexCount, index) ? indexCount[val] += 1 : indexCount[val] = 1;
-      });
-      var bookContainsAll = [];
-      /* Knowing that each index can only occur once for a term
-       if an index frequency is equal to the number of search terms
-       then each search term has the index
-       ===> the book at that index has all terms */
-      Object.keys(indexCount).forEach(function (bookIndex) {
-        if (indexCount[bookIndex] === termsArray.length) {
-          bookContainsAll.push(bookIndex);
-        }
-      });
-      return bookContainsAll;
-    }
-  }]);
-
-  return InvertedIndex;
-}();
-
-exports.default = InvertedIndex;
```

Comment:
Yes, I have added it to .gitignore already but it's already being tracked. I will make neccesary corrections.